### PR TITLE
Add real-world examples using the task schedule

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: [1755](https://github.com/ballerina-platform/ballerina-standard-library/issues/1755)

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by MadhukaHarith92 — DISMISSED** [review]
LGTM

**Review by praneesha — DISMISSED** [review]

File: examples/closed-issue-counter/job.bal [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            io:println("Closed issue count details on " + dateInString + ":\n" + colsedIssueCounts);
+        } else {
+            log:printError("Failed to get the closed issue count details", 'error = colsedIssueCounts);
+        }
+    }
+
+    isolated function init() {}
+}
+
+function getColsedIssueCounts(string[] entries, string date) returns string {
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:60]:
```suggestion
function getClosedIssueCounts(string[] entries, string date) returns string {
```

File: examples/closed-issue-counter/modules/github/client.bal [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import ballerina/http;
```
Comment (daneshk) [inline examples/closed-issue-counter/modules/github/client.bal:1]: License headers?

File: examples/closed-issue-counter/job.bal [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+github:Configuration config = {
+    accessToken: github_access_token
+};
+
+class Job {
+
+    *task:Job;
+
+    public function execute() {
+        io:println("Job started to executed.");
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:37]: shall we use log instead of io:println

File: examples/closed-issue-counter/job.bal [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        string months = civil.month.toString();
+        if months.length() == 1 {
+            months = "0" + months;
+        }
+        string days = civil.day.toString();
+        if days.length() == 1 {
+            days = "0" + days;
+        }
+        string dateInString = (civil.year).toString() + "-" + months + "-" + days;
+        string|error colsedIssueCounts = getClosedIssueCounts(repoNamesWithLabel, dateInString);
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:49]:
```suggestion
        string|error closedIssueCounts = getClosedIssueCounts(repoNamesWithLabel, dateInString);
```

File: examples/closed-issue-counter/job.bal [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        if colsedIssueCounts is string {
+            io:println("Closed issue count details on " + dateInString + ":\n" + colsedIssueCounts);
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:51]:
```suggestion
        if closedIssueCounts is string {
            io:println("Closed issue count details on " + dateInString + ":\n" + closedIssueCounts);
```

Comment (daneshk) [reply examples/closed-issue-counter/job.bal:51]: please use log here

File: examples/closed-issue-counter/job.bal [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            log:printError("Failed to get the closed issue count details", 'error = colsedIssueCounts);
+        }
+    }
+
+    isolated function init() {}
+}
+
+function getColsedIssueCounts(string[] entries, string date) returns string {
+    string totalCount = "";
+    github:Client|error githubClient = new (config);
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:62]: Shall we initialize the client at module level

File: examples/closed-issue-counter/modules/github/client.bal [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    if labelName is string {
+       query = string `is:${state} is:issue label:${labelName} closed:${date}..${date} repo:${repositoryOwnerName}/${repositoryName}`;
+    } else {
+       query = string `is:${state} is:issue closed:${date}..${date} repo:${repositoryOwnerName}/${repositoryName}`;
+    }
+    return "{\"query\": \"query {\n" +
+           "search(query: \\\"" + query + "\\\", type: ISSUE, first: 100) {\n" +
+           "   issueCount\n" +
+           "   }\n" +
+           "}\"}";
```
Comment (daneshk) [inline examples/closed-issue-counter/modules/github/client.bal:65]: shall we create a json value directly instead of string value

Comment (daneshk) [reply examples/closed-issue-counter/modules/github/client.bal:65]: https://ballerina.io/learn/by-example/working-directly-with-json

File: examples/closed-issue-counter/modules/github/client.bal [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    http:Request request = new;
+    request.setHeader("Authorization", "token " + accessToken);
+    json convertedQuery = check stringQuery.fromJsonString();
+
+    request.setJsonPayload(convertedQuery);
+    http:Response response = check graphQlClient->post(EMPTY_STRING, request);
+    return check response.getJsonPayload();
+}
+
+isolated function getFormulatedStringQueryForGetIssueList(string repositoryOwnerName, string repositoryName,
```
Comment (daneshk) [inline examples/closed-issue-counter/modules/github/client.bal:53]:
```suggestion
isolated function getIssueListQueryJson(string repositoryOwnerName, string repositoryName,
```

File: examples/closed-issue-counter/closed_issue_counter.bal [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        string timeValue = (check startTime.startTime).toString();
+        if timeValue != "" {
+            task:JobId jobId = check task:scheduleJobRecurByFrequency(new Job(), interval, repeatingCount,
+            startTime = check time:civilFromString(timeValue));
+            return jobId.id;
+        } else {
+            task:JobId jobId = check task:scheduleJobRecurByFrequency(new Job(), interval, repeatingCount);
+            return jobId.id;
+        }
+    }
```
Comment (daneshk) [inline examples/closed-issue-counter/closed_issue_counter.bal:48]: Shall we combine these two resources together by adding repeatingCount to payload.

File: examples/closed-issue-counter/job.bal [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+                if orgname.length() > 1 {
+                    totalCount += orgname[0] + ":" + orgname[1] + " " + value.toString() + "\n";
+                } else {
+                    totalCount += orgname[0] + " " + value.toString() + "\n";
+                }
+            } else {
+                log:printError("Error:", value);
+            }
+        } else {
+            log:printError("Error:", issueCount);
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:83]: shall we return the error by `check` and handle it inside execute function

File: examples/closed-issue-counter/job.bal [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+    return totalCount;
+}
```
Comment (daneshk) [inline examples/closed-issue-counter/job.bal:87]:
```suggestion
function getClosedIssueCount(string entry, string date) returns int|error {
    string[] orgname = regex:split(entry, ":");
    if orgname.length() > 1 {
        json issueCount = check githubClient->getRepositoryIssueList(repositoryOwner, orgname[0], date, orgname[1]);
        return check <int>issueCount.data.search.issueCount;
    } else {
        json issueCount = check githubClient->getRepositoryIssueList(repositoryOwner, orgname[0], date);
        return check <int>issueCount.data.search.issueCount;
    }
}
```

Comment (daneshk) [reply examples/closed-issue-counter/job.bal:87]: shall we simplify the function like below, the process only one entry. Iterate the entries inside execute function and print log message.

File: examples/closed-issue-counter/modules/github/client.bal [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+    string accessToken;
+    http:Client githubGraphQlClient;
+
+    public isolated function init(Configuration config) returns error? {
+        self.accessToken = config.accessToken;
+        self.githubGraphQlClient = check new(GIT_GRAPHQL_API_URL, config.clientConfig);
+    }
+
+    remote isolated function getRepositoryIssueList(string repositoryOwnerName, string repositoryName,
```
Comment (daneshk) [inline examples/closed-issue-counter/modules/github/client.bal:32]:
```suggestion
    remote isolated function getClosedIssueList(string repositoryOwnerName, string repositoryName,
```

Comment (daneshk) [reply examples/closed-issue-counter/modules/github/client.bal:32]: Shall we rename the function to `getClosedIssueList`, as it always return closed issue list

File: examples/closed-issue-counter/modules/github/client.bal [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+                                                    string date, string? label = ()) returns json|error {
+        return getRepositoryIssueList(repositoryOwnerName, repositoryName, date,
+                        self.accessToken, self.githubGraphQlClient, label);
+    }
+}
+
+isolated function getRepositoryIssueList(string repositoryOwnerName, string repositoryName,
```
Comment (daneshk) [inline examples/closed-issue-counter/modules/github/client.bal:39]:
```suggestion
isolated function getClosedIssueList(string repositoryOwnerName, string repositoryName,
```

Comment (daneshk) [reply examples/closed-issue-counter/modules/github/client.bal:39]: same here

File: examples/closed-issue-counter/closed_issue_counter.bal [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+import ballerina/lang.'int as integer;
+import ballerina/task;
+import ballerina/time;
+import ballerina/jballerina.java;
+
+listener http:Listener issueCounter = new (9092);
+
+service /scheduler on issueCounter {
+
+    resource function post recurJob/[decimal interval](@http:Payload json config) returns error|int {
```
Comment (daneshk) [inline examples/closed-issue-counter/closed_issue_counter.bal:27]: shall we read payload as a record type, like
```
type JobOptions record {
   string startTime?;
   string repeatingCount?;
}
```
and read as,

```
string? startTime = options?.startTime;
```

File: examples/closed-issue-counter/closed_issue_counter.bal [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+        string startTime = (check config.startTime).toString();
+        string repeatingCount = (check config.repeatingCount).toString();
```
Comment (daneshk) [inline examples/closed-issue-counter/closed_issue_counter.bal:29]: what if the payload doesn't contain those two fields

Comment (kalaiyarasiganeshalingam) [reply examples/closed-issue-counter/closed_issue_counter.bal]: Updated the coding by removing if clauses

File: examples/closed-issue-counter/closed_issue_counter.bal [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+   int repeatingCount?;
+};
+
+listener http:Listener issueCounter = new (9092);
+
+service /scheduler on issueCounter {
+
+    resource function post recurJob/[decimal interval](@http:Payload JobOptions config) returns error|int {
+        string? startTime = config["startTime"];
+        int? count = config["repeatingCount"];
```
Comment (daneshk) [inline examples/closed-issue-counter/closed_issue_counter.bal:33]:
```suggestion
        string? startTime = config?.startTime;
        int? count = config?.repeatingCount;
```

**Review by daneshk — DISMISSED** [review]
LGTM

**Review by daneshk — APPROVED** [review]

**Pull request merged**